x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/agent/sub_graphs/translate_rule/nodes/validation/validation.ts

```diff
@@ -25,13 +25,14 @@ export const getValidationNode = ({ logger }: GetValidationNodeParams): GraphNod
     // We want to prevent infinite loops, so we increment the iterations counter for each validation run.
     const currentIteration = ++state.validation_errors.iterations;
     let esqlErrors: string = '';
-    if (!isEmpty(query)) {
-      const { errors, isEsqlQueryAggregating, hasMetadataOperator } = parseEsqlQuery(query);
+    const sanitizedQuery = query ? removePlaceHolders(query) : '';
```

Comment:
won't queries like
1. `|`
2. `[macro:something]`
3. `[lookup:something]`
and similar variations, lead to an empty `sanitizedQuery` after `removePlaceHolders(query)` call? And this will suggest those queries pass ES|QL validation?

Reply:
Yes, that's fine, the graph tries to fix queries that do not validate here. I don't think we should try to fix queries like these. They will be labeled as _partial_ translation at the last `translationResult` node.